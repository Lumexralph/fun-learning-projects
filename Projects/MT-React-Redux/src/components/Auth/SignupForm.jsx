import React from 'react';
import PropTypes from 'prop-types';

const SignupForm = ({
  onSubmit,
  onUsernameChange,
  onEmailChange,
  onPasswordChange,
}) => (
  <div className="right-part">
    <form onSubmit={event => onSubmit(event)}>
      <div className="container">
        <div className="popup">
          <span className="popuptext" id="myPopup" />
        </div>
        <label htmlFor="username">
          <input
            onChange={event => onUsernameChange(event)}
            type="text"
            placeholder="pick a username"
            name="username"
            pattern="[a-zA-Z]+"
            title="Please, provide username with only letters."
            required
          />
        </label>


        <label htmlFor="email">
          <input
            onChange={event => onEmailChange(event)}
            type="email"
            placeholder="your email address"
            name="email"
            required
          />
        </label>


        <label htmlFor="psw1">
          <input
            onChange={event => onPasswordChange(event)}
            type="password"
            placeholder="create a password"
            name="psw1"
            minLength="8"
            pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$"
            title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
            required
          />
        </label>


        <label htmlFor="psw2">
          <input
            onChange={event => onPasswordChange(event)}
            type="password"
            placeholder="confirm password"
            name="psw2"
            minLength="8"
            pattern="^(?=.*\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\s).*$"
            title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
            required
          />
        </label>

        <p>Use at least 1 uppercase, 1 lowercase character, and 1 number</p>

        <button
          className="btn signup-btn background-tertiary"
          type="submit"
        >Sign up for FixZit
        </button>
      </div>
    </form>
  </div>
);

SignupForm.propTypes = {
  onUsernameChange: PropTypes.func.isRequired,
  onEmailChange: PropTypes.func.isRequired,
  onPasswordChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
};


export default SignupForm;
