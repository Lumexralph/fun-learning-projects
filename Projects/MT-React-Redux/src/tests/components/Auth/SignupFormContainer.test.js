import React from 'react';
import renderer from 'react-test-renderer';

import SignupForm from '../../../components/Auth/SignupFormContainer';

describe('Test for Signup Form container', () => {
  const component = renderer.create(
    <SignupForm
      onSubmit={value => value}
      onUsernameChange={value => value}
      onEmailChange={e => e}
      onPasswordChange={e => e}
    />,
  );

  const tree = component.toJSON();
  it('Should render properly', () => {
    expect(tree).toMatchSnapshot();
  });
});
