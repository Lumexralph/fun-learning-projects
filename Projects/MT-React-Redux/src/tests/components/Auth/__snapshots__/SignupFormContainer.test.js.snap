// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Signup Form container Should render properly 1`] = `
<main
  className="signup-container-img"
>
  <section
    className="signup-container"
  >
    <div
      className="right-part"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="container"
        >
          <div
            className="popup"
          >
            <span
              className="popuptext"
              id="myPopup"
            />
          </div>
          <label
            htmlFor="username"
          >
            <input
              name="username"
              onChange={[Function]}
              pattern="[a-zA-Z]+"
              placeholder="pick a username"
              required={true}
              title="Please, provide username with only letters."
              type="text"
            />
          </label>
          <label
            htmlFor="email"
          >
            <input
              name="email"
              onChange={[Function]}
              placeholder="your email address"
              required={true}
              type="email"
            />
          </label>
          <label
            htmlFor="psw1"
          >
            <input
              minLength="8"
              name="psw1"
              onChange={[Function]}
              pattern="^(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\\\\s).*$"
              placeholder="create a password"
              required={true}
              title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
              type="password"
            />
          </label>
          <label
            htmlFor="psw2"
          >
            <input
              minLength="8"
              name="psw2"
              onChange={[Function]}
              pattern="^(?=.*\\\\d)(?=.*[a-z])(?=.*[A-Z])(?!.*\\\\s).*$"
              placeholder="confirm password"
              required={true}
              title="Please include at least 1 uppercase character, 1 lowercase character, and 1 number."
              type="password"
            />
          </label>
          <p>
            Use at least 1 uppercase, 1 lowercase character, and 1 number
          </p>
          <button
            className="btn signup-btn background-tertiary"
            type="submit"
          >
            Sign up for FixZit
          </button>
        </div>
      </form>
    </div>
  </section>
</main>
`;
